plugins/woocommerce/includes/class-wc-product-variation.php

```diff
@@ -572,7 +573,7 @@ public function variation_is_active() {
 	 * @return bool
 	 */
 	public function variation_is_visible() {
-		return apply_filters( 'woocommerce_variation_is_visible', 'publish' === get_post_status( $this->get_id() ) && '' !== $this->get_price(), $this->get_id(), $this->get_parent_id(), $this );
+		return apply_filters( 'woocommerce_variation_is_visible', ProductStatus::PUBLISH === get_post_status( $this->get_id() ) && '' !== $this->get_price(), $this->get_id(), $this->get_parent_id(), $this );
```

Comment:
To satisfy the linter, we may wish to document this hook (even though the hook itself is not new).